File: MyListView/app/src/main/java/com/example/mylistview/MainActivity.java
package com.example.mylistview;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ListView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ListView listView = (ListView) findViewById(R.id.listView);

        SingerAdapter adapter = new SingerAdapter();
        adapter.addItem(new SingerItem("소녀시대", "010125325222"));
        adapter.addItem(new SingerItem("녀시대", "010125325222"));
        adapter.addItem(new SingerItem("시대", "010125325222"));
        adapter.addItem(new SingerItem("대", "010125325222"));
        adapter.addItem(new SingerItem("소녀시", "010125325222"));
        adapter.addItem(new SingerItem("소녀", "010125325222"));
        adapter.addItem(new SingerItem("소시대", "010125325222"));
        listView.setAdapter(adapter);

    }

    class SingerAdapter extends BaseAdapter{

        ArrayList<SingerItem> items = new ArrayList<SingerItem>();
        @Override
        public int getCount() {
            return items.size();
        }

        public void addItem(SingerItem item){
            items.add(item);
        }

        @Override
        public Object getItem(int i) {
            return items.get(i);
        }

        @Override
        public long getItemId(int i) {
            return i;
        }

        @Override
        public View getView(int i, View convertview, ViewGroup viewGroup) {
            SingerItemView view = new SingerItemView(getApplicationContext());
            SingerItem item = items.get(i);
            view.setName(item.getName());
            view.setMobile(item.getMobile());

            return view;
        }
    }
}

File: MyRecyclerViewKt/app/src/main/java/com/example/recyclerviewkt/MainActivity.kt
package com.example.recyclerviewkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.provider.ContactsContract
import android.widget.LinearLayout
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val profileList = arrayListOf(
            Profiles(R.drawable.man, "이성찬", 26, "개발자"),
            Profiles(R.drawable.woman, "조형진", 27, "안드로이드 개발자"),
            Profiles(R.drawable.woman, "이주현", 23, "하이브리드 개발자"),
            Profiles(R.drawable.man, "김세정", 24, "웹 앱 개발자"),
            Profiles(R.drawable.woman, "이동주", 21, "리액트 개발자"),
            Profiles(R.drawable.man, "김병윤", 30, "그냥 개발자"),
            Profiles(R.drawable.man, "김나영", 35, "배고픈 개발자"),
            Profiles(R.drawable.woman, "이승주", 42, "유니티 개발자"),
            Profiles(R.drawable.man, "조병재", 23, "클린 개발자"),
            Profiles(R.drawable.man, "이아영", 26, "아이폰 개발자"),
            Profiles(R.drawable.woman, "이나현", 19, "개발자"),
            Profiles(R.drawable.woman, "조예빈", 26, "개발자")
        )

        rv_profile.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        rv_profile.setHasFixedSize(true)

        rv_profile.adapter = ProfileAdapter(profileList);
    }

}
